src/main/java/com/PBA/authservice/controller/request/UserUpdateRequest.java

```diff
@@ -0,0 +1,43 @@
+package com.pba.authservice.controller.request;
+
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import jakarta.validation.constraints.Size;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+public class UserUpdateRequest {
+    @NotBlank(message = "{username.notblank}")
+    @Size(min = 4, message = "{username.minsize}")
+    @Size(max = 20, message = "{username.maxsize}")
+    private String username;
+
+    @NotBlank(message = "{password.notblank}")
+    @Size(min = 7, message = "{password.minsize}")
+    @Size(max = 20, message = "{password.maxsize}")
+    @Pattern(regexp = ".*[!@#$%^&*?].*", message = "{password.specialchar}")
+    @Pattern(regexp = ".*[A-Z].*", message = "{password.uppercase}")
+    private String password;
```

Comment:
The password can't be updated here, we will have a functionality for forgot password.